crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -1513,28 +1550,35 @@ impl<'db> TypeInferenceBuilder<'db> {
         // If the target defines, e.g., `__iadd__`, infer the augmented assignment as a call to that
         // dunder.
         if let Type::Instance(class) = target_type {
-            let class_member = class.class_member(self.db, op.in_place_dunder());
-            if !class_member.is_unbound() {
-                let call = class_member.call(self.db, &[target_type, value_type]);
-                return match call.return_ty_result(
-                    self.db,
-                    AnyNodeRef::StmtAugAssign(assignment),
-                    &mut self.diagnostics,
-                ) {
-                    Ok(t) => t,
-                    Err(e) => {
-                        self.diagnostics.add(
-                            assignment.into(),
-                            "unsupported-operator",
-                            format_args!(
-                                "Operator `{op}=` is unsupported between objects of type `{}` and `{}`",
-                                target_type.display(self.db),
-                                value_type.display(self.db)
-                            ),
-                        );
-                        e.return_ty()
-                    }
-                };
+            match class.class_member(self.db, op.in_place_dunder()) {
+                SymbolLookupResult::Unbound => {}
+                SymbolLookupResult::Type(class_member, _boundedness) => {
+                    // TODO: Handle the case where boundness is `MayBeUnbound`: fall back
+                    // to the binary-op behavior below and union the result with calling
+                    // the potentially-unbound in-place dunder.
+
+                    let call = class_member.call(self.db, &[target_type, value_type]);
+
+                    return match call.return_ty_result(
+                        self.db,
+                        AnyNodeRef::StmtAugAssign(assignment),
+                        &mut self.diagnostics,
+                    ) {
+                        Ok(t) => t,
+                        Err(e) => {
+                            self.diagnostics.add(
+                                assignment.into(),
+                                "unsupported-operator",
+                                format_args!(
+                                    "Operator `{op}=` is unsupported between objects of type `{}` and `{}`",
+                                    target_type.display(self.db),
+                                    value_type.display(self.db)
+                                ),
+                            );
+                            e.return_ty()
+                        }
+                    };
+                }
```

Comment:
Nit: The diff here could be smaller if you used an `if let SymbolLookupResult::Type(..)` or in combination with an `as_type` method

```
if let Some(class_member) = class.class_member(self.db, op.in_place_dunder()).as_type()) {
	...
}
```

Reply:
Thanks!